pkg/steps/multi_stage.go

```diff
@@ -172,14 +176,29 @@ func (s *multiStageTestStep) run(ctx context.Context) error {
 		return fmt.Errorf("failed to create RBAC objects: %w", err)
 	}
 	var errs []error
+	observers, err := s.generateObservers(s.observers)
+	if err != nil {
+		// if we can't even generate the Pods there's no reason to run the job
+		return err
+	}
+	observerContext, cancel := context.WithCancel(ctx)
+	observerDone := make(chan interface{})
```

Comment:
Minor point, but did you mean `struct{}`?  Even empty interfaces are a data/vtable pointer pair:

![](http://research.swtch.com/gointer3.png)

https://research.swtch.com/interfaces